[collector] Update internal metrics config examples, suggest OTLP for pushing metrics (#5212)
open-telemetry/opentelemetry.io: content/en/docs/collector/internal-telemetry.md

You can configure how internal metrics are generated and exposed by the
Collector. By default, the Collector generates basic metrics about itself and

> exposes them using the otel-go Prometheus exporter for scraping the at

Reviewer: ```suggestion
exposes them using the OpenTelemetry Go [Prometheus exporter](https://github.com/open-telemetry/opentelemetry-go/tree/main/exporters/prometheus) for scraping the at
```

You can configure how internal metrics are generated and exposed by the
Collector. By default, the Collector generates basic metrics about itself and
exposes them using the OpenTelemetry Go
[Prometheus exporter](https://github.com/open-telemetry/opentelemetry-go/tree/main/exporters/prometheus)

> for scraping the at `http://127.0.0.1:8888/metrics`. You can expose the endpoint

Reviewer 2: ```suggestion
for scraping at `http://127.0.0.1:8888/metrics`. You can expose the endpoint
```

      level: detailed
```

The Collector can also be configured to push its own metrics to an
[OTLP receiver](https://github.com/open-telemetry/opentelemetry-collector/tree/main/receiver/otlpreceiver)
and send them through configured pipelines. In the following example,
the Collector is configured to push metrics every 10s using OTLP gRPC to

> `localhost:8888`:

Reviewer: ```suggestion
`localhost:14317`:
```

```yaml
receivers:
  otlp/internal_metrics:
    protocols:
      grpc:

>         endpoint: localhost:8888

Reviewer: ```suggestion
        endpoint: localhost:14317
```

exporters:
  debug:
service:
  pipelines:
    metrics:
      receivers: [otlp/internal_metrics]
      exporters: [debug]
  telemetry:
    metrics:
      readers:
        - periodic:
            interval: 10000
            exporter:
              otlp/metrics-agent:
                protocol: grpc/protobuf

>                 endpoint: localhost:8888

Reviewer: ```suggestion
                endpoint: localhost:14317
```
